### PR TITLE
chore: knowledgebase add processing status

### PR DESCRIPTION
**Pull request by 0xff-dev**
#### What this PR does / why we need it
chore: knowledgebase add processing status

#### Which issue(s) this PR fixes

Fix https://github.com/kubeagi/arcadia/issues/922

### CODE REVIEW

File: controllers/base/knowledgebase_controller.go [hunk 1]
```diff
@@ -63,6 +63,9 @@ const (
 	waitLonger  = time.Hour
 	waitSmaller = time.Second * 3
 	waitMedium  = time.Minute
+
+	retryForAll    = "for-all"
```
Comment (bjwswang) [inline controllers/base/knowledgebase_controller.go:67]: when will you use this const? retryForAll

File: controllers/base/knowledgebase_controller.go [hunk 2]
```diff
@@ -139,23 +128,78 @@ func (r *KnowledgeBaseReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		return ctrl.Result{}, nil
 	}
 
+	// Add a finalizer.Then, we can define some operations which should
+	// occur before the KnowledgeBase to be deleted.
```
Comment (bjwswang) [inline controllers/base/knowledgebase_controller.go:132]: `before` or `after`

Comment (0xff-dev) [reply controllers/base/knowledgebase_controller.go:132]: `Before`. The finalizer is used to clean up the resource before it is actually deleted.

**Review by bjwswang — APPROVED** [review]

**Comment by bjwswang** [comment]
@0xff-dev please work with @Carrotzpc on these new updates.

**Pull request merged**